src/docfx/build/Build.cs

```diff
@@ -91,40 +91,20 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Repo
                 ValidateBookmarks();
 
                 var manifest = manifestBuilder.Build(context);
-                var allDependencies = sourceDependencies.OrderBy(d => d.Key.FilePath).ToDictionary(k => k.Key, v => v.Value);
-                var allDependencyMap = new DependencyMap(allDependencies);
+                var dependencyMap = dependencyMapBuilder.Build();
 
-                return (CreateManifest(manifest, allDependencyMap), manifest, allDependencyMap);
+                return (CreateManifest(manifest, dependencyMap), manifest, dependencyMap);
 
                 async Task<List<string>> BuildOneFile(Document file, Action<Document> buildChild)
                 {
-                    var dependencyMapBuilder = new DependencyMapBuilder();
                     var callback = new PageCallback(xrefMap, dependencyMapBuilder, bookmarkValidator, buildChild);
-                    var (dependencyMap, monikers) = await BuildFile(context, file, tocMap, contribution, null, callback, manifestBuilder);
-
-                    foreach (var (source, dependencies) in dependencyMap)
-                    {
-                        sourceDependencies.TryAdd(source, dependencies);
-                    }
-
-                    return monikers;
+                    return await BuildFile(context, file, tocMap, contribution, null, callback, manifestBuilder);
                 }
 
                 async Task BuildTocFile(Document file, Dictionary<Document, List<string>> map)
                 {
-                    var (dependencyMap, _) = await BuildFile(
-                        context,
-                        file,
-                        tocMap,
-                        contribution,
-                        map,
-                        null,
-                        manifestBuilder);
-
-                    foreach (var (source, dependencies) in dependencyMap)
-                    {
-                        sourceDependencies.TryAdd(source, dependencies);
-                    }
+                    var callback = new PageCallback(xrefMap, dependencyMapBuilder, bookmarkValidator, null);
```

Comment:
>var callback = new PageCallback(xrefMap, dependencyMapBuilder, bookmarkValidator, null); [](start = 20, length = 88)

when build TOC, we don't need that callback, it's confused me that what should i put there, why not just leave callback to buildPage and just pass the dependencyBuilder to toc

Reply:
Maybe we should expand `callback`, here to build TOC, we only need to send in the global `dependencyMapBuilder`

Reply:
This is the first iteration of the refactor to make changes small. `DependencyMapBuild`, `PageCallback`, `Resolve` will be consolidated to a single `DependencyResolver` that is responsible for resolve content, url, xref dependencies. TOC also need to resolve xref so it depend on `DependencyResolver` hence I am putting it here.